app/gui/view/execution-mode-selector/src/lib.rs

```diff
@@ -0,0 +1,250 @@
+//! UI component that allows selecting a execution mode.
+
+#![recursion_limit = "512"]
+// === Features ===
+#![feature(option_result_contains)]
+#![feature(trait_alias)]
+// === Standard Linter Configuration ===
+#![deny(non_ascii_idents)]
+#![warn(unsafe_code)]
+#![allow(clippy::bool_to_int_with_if)]
+#![allow(clippy::let_and_return)]
+// === Non-Standard Linter Configuration ===
+#![warn(missing_copy_implementations)]
+#![warn(missing_debug_implementations)]
+#![warn(missing_docs)]
+#![warn(trivial_casts)]
+#![warn(trivial_numeric_casts)]
+#![warn(unused_import_braces)]
+#![warn(unused_qualifications)]
+
+use enso_prelude::*;
+use ensogl::prelude::*;
+
+use enso_frp as frp;
+use ensogl::application::Application;
+use ensogl::data::color::Rgba;
+use ensogl::display;
+use ensogl::display::camera::Camera2d;
+use ensogl_gui_component::component;
+use ensogl_hardcoded_theme::graph_editor::execution_mode_selector as theme;
+
+
+
+// ================
+// === Constants ===
+// ================
+
+const PLAY_BUTTON_SIZE: f32 = 10.0;
+const PLAY_BUTTON_OFFSET: f32 = 15.0;
+const PLAY_BUTTON_PADDING: f32 = 10.0;
+const DIVIDER_OFFSET: f32 = 32.5;
+const DIVIDER_PADDING: f32 = 10.0;
+const DROPDOWN_WIDTH: f32 = 95.0;
+/// Width of the execution mode selector.
+pub const OVERALL_WIDTH: f32 =
+    DROPDOWN_WIDTH + 2.0 * DIVIDER_PADDING + PLAY_BUTTON_SIZE + PLAY_BUTTON_PADDING;
+/// Height of the execution mode selector.
+pub const HEIGHT: f32 = 24.0;
+
+
+// ==============
+// === Shapes ===
+// ==============
+
+mod background {
+    use super::*;
+
+    ensogl::shape! {
+        below = [play_icon];
+        (style:Style) {
+            let size = Var::canvas_size();
+            let height = size.y();
+            let width = size.x();
+            let color = style.get_color(theme::background);
+            let base_rect = Rect(size).corners_radius(&height/2.px()).fill(color);
+
+            let divider = Rect((1.px(), &height)).fill(style.get_color(theme::divider));
+            let divider = divider.translate_x(width/2.0 - DIVIDER_OFFSET.px());
+
+            (base_rect + divider).into()
+        }
+    }
+}
```

Comment:
Please use our super-box instead. You can find it in `lib/rust/ensogl/core/src/display/shape/compound/rectangle.rs`. It should be used whenever we can to not define new shapes and thus, to maintain the same number of draw calls. It's OK to use several rectangles - one for divider, one for left part and one for right part.